apps/condo/domains/common/utils/text.js

```diff
@@ -0,0 +1,44 @@
+function removeSpacesInsideQuotes (s) {
+    const nestedQuotes = []
+    let res = [...s]
+    for (let i = 0; i < res.length; i++) {
+        const char = res[i]
+        if (char === '"') {
+            if (nestedQuotes[nestedQuotes.length - 1] === char) {
+                nestedQuotes.pop()
+                if (res[i - 1] === ' ') res[i - 1] = ''
+            } else {
+                nestedQuotes.push(char)
+                if (res[i + 1] === ' ') res[i + 1] = ''
+            }
+        }
+    }
+    return res.join('')
+}
+
+function normalizeText (text) {
+    if (!text) return
+    String(text).normalize()
+    const EOL = text.match(/\r\n/gm) ? '\r\n' : '\n'
+    const threeOrMoreEOL = new RegExp(`(${EOL}){3,}`, 'gm')
+    const punctuations = '.,:'
+    const formatted = text
+        // replace three or more EOL to two EOL
+        .replace(threeOrMoreEOL, EOL + EOL)
+        // replace two or more spaces to one space
+        .replace(/ {2,}/gm, ' ')
```

Comment:
This regex will not capture this sample `lorem  ipsum` copy-paste.
It's because it contains non-breaking whitespaces ;)

Lines in suggested code below does't look the same;)
1. First line replaces regular white-spaces
2. Second line replaces non-breaking white-spaces
3. Third line replaces regular or non-breaking white-space into one non-breaking white-space

```suggestion
        .replace(/ {2,}/gm, ' ')
        .replace(/ {2,}/gm, ' ')
        .replace(/[  ]{2,}/gm, ' ')
```